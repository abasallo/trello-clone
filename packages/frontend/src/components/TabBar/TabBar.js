import React, { useState } from 'react'

import Tabs from '@material-ui/core/Tabs'
import Tab from '@material-ui/core/Tab'

import TabPanel from '../TabPanel/TabPanel'
import CellMaps from '../CellMaps/CellMaps'
import Copyright from '../Copyright/Copyright'

import { Container, CopyrightBox } from './TabBar.styled.components'

const TabBar = (props) => {
  const [currentTab, setCurrentTab] = useState(0)

  const handleChange = (event, value) => setCurrentTab(value)

  return (
    <Container>
      <Tabs value={currentTab} onChange={handleChange} aria-label="simple tabs example">
        <Tab label="Home" id={'simple-tab-0'} aria-controls={'simple-tabpanel-0'} />
        <Tab label="CellMaps with Google Maps" id={'simple-tab-1'} aria-controls={'simple-tabpanel-1'} />
        <Tab label="Chat Bot" id={'simple-tab-2'} aria-controls={'simple-tabpanel-2'} />
        <Tab label="DB Connection Test" id={'simple-tab-3'} aria-controls={'simple-tabpanel-3'} />
      </Tabs>

      <TabPanel value={currentTab} index={0}></TabPanel>

      <TabPanel value={currentTab} index={1}>
        <CellMaps></CellMaps>
      </TabPanel>

      <TabPanel value={currentTab} index={2} />

      <TabPanel value={currentTab} index={3}>
        <h1>Names</h1>
        <p>{props.user.names}</p>
        <h1>Surnames</h1>
        <p>{props.user.surnames}</p>
        <h1>Email</h1>
        <p>{props.user.email}</p>
      </TabPanel>

      <CopyrightBox mt={8}>
        <Copyright />
      </CopyrightBox>
    </Container>
  )
}

export default TabBar
