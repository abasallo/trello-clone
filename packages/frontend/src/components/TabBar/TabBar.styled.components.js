import styled from 'styled-components'

import MaterialUIContainer from '@material-ui/core/Container'
import Box from '@material-ui/core/Box'

import { ChatBot as AmplifyChatBot } from 'aws-amplify-react'

export const Container = styled(MaterialUIContainer)`
  margin-top: 25px;
  text-align: center;
`

export const ChatBot = styled(AmplifyChatBot)`
  margin-bottom: 25px;
`

export const CopyrightBox = styled(Box)`
  margin-top: 25px;
  text-align: center;
`
