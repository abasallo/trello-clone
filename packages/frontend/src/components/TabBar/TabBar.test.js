import React from 'react'
import renderer from 'react-test-renderer'

import TabBar from './TabBar'

test('Renders properly', () =>
  expect(renderer.create(<TabBar user={{ names: 'names', surnames: 'surnames', email: 'email' }} />).toJSON()).toMatchSnapshot())
