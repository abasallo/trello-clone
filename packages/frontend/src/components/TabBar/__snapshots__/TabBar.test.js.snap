// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<div
  className="MuiContainer-root sc-bdnylx hXKTtf MuiContainer-maxWidthLg"
>
  <div
    className="MuiTabs-root"
  >
    <div
      className="MuiTabs-scroller MuiTabs-fixed"
      onScroll={[Function]}
      style={
        Object {
          "marginBottom": null,
          "overflow": "hidden",
        }
      }
    >
      <div
        aria-label="simple tabs example"
        className="MuiTabs-flexContainer"
        onKeyDown={[Function]}
        role="tablist"
      >
        <button
          aria-controls="simple-tabpanel-0"
          aria-selected={true}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
          disabled={false}
          id="simple-tab-0"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Home
          </span>
          <span
            className="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
            style={Object {}}
          />
        </button>
        <button
          aria-controls="simple-tabpanel-1"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-1"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            CellMaps with Google Maps
          </span>
        </button>
        <button
          aria-controls="simple-tabpanel-2"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-2"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            Chat Bot
          </span>
        </button>
        <button
          aria-controls="simple-tabpanel-3"
          aria-selected={false}
          className="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
          disabled={false}
          id="simple-tab-3"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="tab"
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiTab-wrapper"
          >
            DB Connection Test
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    aria-labelledby="simple-tab-0"
    hidden={false}
    id="simple-tabpanel-0"
    role="tabpanel"
  >
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      />
    </div>
  </div>
  <div
    aria-labelledby="simple-tab-1"
    hidden={true}
    id="simple-tabpanel-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-2"
    hidden={true}
    id="simple-tabpanel-2"
    role="tabpanel"
  />
  <div
    aria-labelledby="simple-tab-3"
    hidden={true}
    id="simple-tabpanel-3"
    role="tabpanel"
  />
  <div
    className="MuiBox-root MuiBox-root-6 sc-dlnjPT eLntHy"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      Copyright Â© 
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
        href="https://ecs.co.uk/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        ECS
      </a>
       2020
    </p>
  </div>
</div>
`;
